src/tests/tracing/eventpipe/common/IpcEndpointConfig.cs

```diff
@@ -0,0 +1,144 @@
+
+            switch (transportType)
+            {
+                case TransportType.NamedPipe:
+                {
+                    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"{NamedPipeSchema} is only supported on Windows.");
+                    break;
+                }
+                case TransportType.UnixDomainSocket:
+                {
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"{UnixDomainSocketSchema} is not supported on Windows, use {NamedPipeSchema}.");
+                    break;
+                }
+                case TransportType.TcpClient:
+                {
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        throw new PlatformNotSupportedException($"TcpClient is not supported on Windows, use {NamedPipeSchema}."); // It is probably supported, just not recommended
+                    break;
+                }
+                default:
+                {
+                    throw new NotSupportedException($"{transportType} not supported.");
+                }
+            }
+
+            Address = address;
+            _transportType = transportType;
+        }
+
+        // Config format: [Address]
+        //
+        // Address in UnixDomainSocket formats:
+        // myport => myport
+        // uds:myport => myport
+        // /User/mrx/myport.sock => /User/mrx/myport.sock
+        // uds:/User/mrx/myport.sock => /User/mrx/myport.sock
+        // uds://authority/User/mrx/myport.sock => /User/mrx/myport.sock
+        // uds:///User/mrx/myport.sock => /User/mrx/myport.sock
+        //
+        // Address in NamedPipe formats:
+        // myport => myport
+        // namedpipe:myport => myport
+        // \\.\pipe\myport => myport (dropping \\.\pipe\ is inline with implemented namedpipe client/server)
+        // namedpipe://./pipe/myport => myport (dropping authority and /pipe/ is inline with implemented namedpipe client/server)
+        // namedpipe:/pipe/myport  => myport (dropping /pipe/ is inline with implemented namedpipe client/server)
+        // namedpipe://authority/myport => myport
+        // namedpipe:///myport => myport
+        public static IpcEndpointConfig Parse(string config)
+        {
+            if (string.IsNullOrEmpty(config))
+                throw new FormatException("Missing IPC endpoint config.");
+
+            string address = "";
+            TransportType transportType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? TransportType.NamedPipe : OperatingSystem.IsAndroid() ? TransportType.TcpClient : TransportType.UnixDomainSocket;
+
+            if (!string.IsNullOrEmpty(config))
+            {
+                var parts = config.Split(','); // Can clean up and assume config is just the address
```

Comment:
I had removed PortType, and if we truly dont need it, then we can clean this up with expectation that config is just the address.